clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java

```diff
@@ -974,6 +1011,9 @@ public void run() {
 
                 // Update the current time and handle the latest responses.
                 now = time.milliseconds();
+                if (handleAuthenticationException(now, callsToSend) &&
+                        hardShutdownTimeMs.get() != INVALID_SHUTDOWN_TIME)
```

Comment:
We don't want to exit this background thread until a close has been requested. If we are closing and there is an authentication exception, we want to terminate immediately. If we are not closing, the thread is kept alive so that a future request after credentials are added can succeed. This is tested in the integration test.

Reply:
Hmm, what's the impact of not exiting immediately? It seems like we would not have any responses anyway, so it should happen pretty quickly either way (and all pending requests have been cancelled). Maybe keep it simple?